zkevm-circuits/src/evm_circuit/witness.rs

```diff
@@ -377,7 +379,62 @@ impl std::ops::Index<(RwTableTag, usize)> for RwMap {
         &self.0.get(&tag).unwrap()[idx]
     }
 }
-
+impl RwMap {
+    /// Check rw_counter is continous and starting from 1
+    pub fn check_rw_counter_sanity(&self) {
```

Comment:
In case the RWs do not pass the sanity check, do we want to panic (the `debug_assert_eq` will panic)? Or handle the error before loading the RW table?

Reply:
there are two scenarios that cannot satisfy `check_rw_counter_sanity`. (1)in evm circuit witness is wrong, it is bug so we debug_assert (2) we manually construct test cases for state circuit where rw_counter may not be continuous.

So i only `check_rw_counter_sanity` in evm circuit assignment, and let it fail if error happens